lib/iris/analysis/__init__.py

```diff
@@ -1412,7 +1410,35 @@ def interp_order(length):
 """
 
 
-MEAN = WeightedAggregator('mean', ma.average, lazy_func=biggus.mean)
+def _build_dask_mdtol_function(dask_nanfunction):
+    """
+    Make a wrapped dask statistic function that supports the 'mdtol' keyword.
+
+    'dask_nanfunction' must be a statistical function that handles NaNs as
+    missing data, and which has the basic call signature
+    "dask_nanfunction(data, axis, **kwargs)".
+
+    The returned value is new function operating on dask arrays.
+    It has the call signature "stat(data, axis=-1, mdtol=None, *kwargs)".
+
+    """
+    @wraps(dask_nanfunction)
+    def inner_stat(array, axis=-1, mdtol=None, **kwargs):
+        dask_result = dask_nanfunction(array, axis=axis, **kwargs)
+        if mdtol is None:
+            result = dask_result
+        else:
+            point_counts = np.sum(np.ones(array.shape), axis=axis)
+            point_mask_counts = da.sum(da.isnan(array), axis=axis)
+            masked_point_fractions = (point_mask_counts + 0.0) / point_counts
```

Comment:
Note: the "+ 0.0" ensures a floating divison, as there is no "dask.float".
I could probably add a comment to that effect....